### PR TITLE
Metrics Backend

### PR DESCRIPTION
**Pull request by julia-martin**
Created API endpoints for all metrics routes, and added documentation to the README.

The POST /metrics endpoint validates the `type` against our list of types, and validates the `query_string` by running the query and making sure it has the right columns. If the request fails validation, the API sends a 400 status code.
The other endpoints are pretty much the same as the others.

### CODE REVIEW

**Comment by seanrichardson95** [comment]
Looks great! Good job.

**Review by seanrichardson95 — COMMENTED** [review]
Not sure if this is work for a separate branch/pull request, but shouldn't the GET api/experiments/:id and GET api/flags/:id/experiments return different stuff since we now have multiple means, p-values, etc. for each metric for each experiment?

File: server/controllers/experimentController.js [hunk 1]
```diff
@@ -58,6 +58,9 @@ const editExperiment = async (req, res, next) => {
   const id = req.params.id;
   try {
     const updatedFields = req.body;
+    if (updatedFields.date_ended) {
```
Comment (seanrichardson95) [inline server/controllers/experimentController.js:61]: Looks great!

**Review by calebrs — COMMENTED** [review]
Looks solid to me!

**Comment by julia-martin** [comment]
@seanrichardson95 Yeah I was thinking we could update that later once we implement the statistics stuff

**Pull request merged**